### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by naoman**

### CODE REVIEW

File: sparkstreaming-job/src/main/java/zipkin/sparkstreaming/job/ZipkinSparkStreamingJob.java [hunk 1]
```diff
@@ -80,4 +98,30 @@ static String pathToUberJar() throws UnsupportedEncodingException {
     URL jarFile = ZipkinSparkStreamingJob.class.getProtectionDomain().getCodeSource().getLocation();
     return URLDecoder.decode(jarFile.getPath(), "UTF-8");
   }
+
```
Comment (naoman) [inline sparkstreaming-job/src/main/java/zipkin/sparkstreaming/job/ZipkinSparkStreamingJob.java]: What would be the best place to put these configurations?

File: sparkstreaming/src/main/java/zipkin/sparkstreaming/ElasticsearchHttpSpanStorage.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+public class ElasticsearchHttpSpanStorage implements TraceConsumer, Cloneable {
+
+  String _esUrl = "http://XXX.us-east-1.es.amazonaws.com";
```
Comment (naoman) [inline sparkstreaming/src/main/java/zipkin/sparkstreaming/ElasticsearchHttpSpanStorage.java]: These will be moved to a separate properties class

File: sparkstreaming/src/main/java/zipkin/sparkstreaming/FinagleSpanProcessor.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class FinagleSpanProcessor implements SpanProcessor {
+
+  private static final Logger logger = LoggerFactory.getLogger(FinagleSpanProcessor.class);
+
+  static boolean _dropFinagleFlushAnnotation = true;
```
Comment (naoman) [inline sparkstreaming/src/main/java/zipkin/sparkstreaming/FinagleSpanProcessor.java]: These will be moved to a separate properties class

File: sparkstreaming/src/main/java/zipkin/sparkstreaming/SparkStreamingJob.java [hunk 4]
```diff
@@ -145,16 +155,19 @@ static void streamSpansToStorage(JavaDStream<byte[]> messageStream, TraceConsume
...
         .map(s -> s /** TODO: plugin some transformer */);
 
-    JavaPairDStream<String, Iterable<Span>> traces = validSpans
+    JavaDStream<Span> processedSpans = spanProcessor.process(validSpans);
+
+    JavaPairDStream<String, Iterable<Span>> traces = processedSpans
         .mapToPair(s -> new Tuple2<>(Util.toLowerHex(s.traceIdHigh, s.traceId), s))
         .groupByKey();
 
+    VoidFunction<Iterator<Tuple2<String, Iterable<Span>>>> storeTrace = p -> {
```
Comment (naoman) [inline sparkstreaming/src/main/java/zipkin/sparkstreaming/SparkStreamingJob.java]: looks like there is some merge issue. Will fix in next commit.

**Review by codefromthecrypt — COMMENTED** [review]
discussed offline that I'll help merge in the finagle part of this, which can help sketch out how auto-configuration works.

**Comment by adriancole** [comment]
Last thing to do from here is add (probably an elasticsearch) storage consumer

**Comment by adriancole** [comment]
okie all the changes from here are integrated, I think. thanks @naoman

**Pull request closed without merging**